### PR TITLE
CASA-10103. Fix Chauvenet bug. Remove _clearData(), use reset() instead

### PR DESCRIPTION
**Pull request by dmehring**
Also move StatisticsAlgorithm::setDataProvider() implementation from .h to .tcc

### CODE REVIEW

**Comment by dmehring** [comment]
@mpokorny Please review and merge once CI has completed successfully. Thanks!

File: scimath/Mathematics/FitToHalfStatistics.tcc [hunk 1]
```diff
@@ -330,13 +325,12 @@ void FitToHalfStatistics<CASA_STATP>::setCalculateAsAdded(
 }
 
 CASA_STATD
-void FitToHalfStatistics<CASA_STATP>::_clearData() {
+void FitToHalfStatistics<CASA_STATP>::reset() {
     _doMedAbsDevMed = False;
     StatsData<AccumType> oldStats = copy(_statsData);
```
Comment (mpokorny) [inline scimath/Mathematics/FitToHalfStatistics.tcc]: Isn't the oldStats value unnecessary given your other changes to the method?

**Comment by mpokorny** [comment]
@dmehring, I commented on a change to the FitToHalfStatistics::reset method. I'm not sure you will have gotten a notification; in case you didn't, this message should do it.

**Comment by dmehring** [comment]
I did see it and good catch. I just pushed a new version of that file so hopefully the change will show up...
Thanks

**Comment by dmehring** [comment]
Oops I pushed a nonexistent branch :) Looks like its there now.

**Pull request merged**